haystack/nodes/retriever/_embedding_encoder.py

```diff
@@ -434,11 +438,93 @@ def save(self, save_dir: Union[Path, str]):
         raise NotImplementedError(f"Saving is not implemented for {self.__class__}")
 
 
+class _BedrockEmbeddingEncoder(_BaseEmbeddingEncoder):
+    def __init__(self, retriever: "EmbeddingRetriever"):
+        boto3_import.check()
+
+        # See https://docs.aws.amazon.com/bedrock/latest/userguide/embeddings.html for more details
+        # The maximum input text is 8K tokens and the maximum output vector length is 1536
+        # Bedrock Titan embeddings do not support batch operations
+        self.aws_config = retriever.aws_config
+        self.client = self.initialize_boto3_session().client("bedrock-runtime")
+        self.model: str = next(
+            (m for m in BEDROCK_EMBEDDING_MODELS if m in retriever.embedding_model), "amazon.titan-embed-text-v1"
+        )
```

Comment:
It would be better to raise an error if the model is not support instead of falling back to the titan model.